src/components/LabeledText/AddressLink.tsx

```diff
@@ -10,45 +10,43 @@ import {
   LabeledLinkPropTypes
 } from './LabeledLink'
 
+/**
+ * AddressLink is a LabeledLink for actor addresses.
+ *
+ * When fetchAddress is false (default), both id and address can be
+ * provided, but only the (robust) address will be shown if present
+ *
+ * When fetchAddress is true, only the address prop needs to be provided
+ * (either id or robust) and the missing information will be fetched.
+ */
 export const AddressLink = ({
   id,
   address,
-  label,
-  color,
-  disableLink,
-  hideCopy,
-  hideCopyText,
   shouldTruncate,
   fetchAddress,
-  useNewTabIcon
+  ...labeledLinkProps
 }: AddressLinkProps) => {
   const { explorerUrl } = useEnvironment()
   const { data: gqlAddress } = useAddressQuery({
-    skip: !id || !!address || !fetchAddress,
-    variables: {
-      address: id
-    }
+    skip: !fetchAddress,
+    variables: { address }
```

Comment:
Changed the logic here a bit, guessing this is what you mentioned before :)

Reply:
Yes